### ISSUE TITLE
Telegraf 1.17 has issues reading a valid telegraf.conf file (text format parsing error in line 1: invalid metric name)

### ISSUE THREAD

**Issue by gsperry2011** [influxdata/telegraf#8719]
### Relevant telegraf.conf:

```toml
# Telegraf Configuration
#
# THIS FILE IS MANAGED BY PUPPET
#
[global_tags]
  ip4_addr = "192.168.1.10"
  job = "telegraf"

[agent]
  hostname = "mytest.system.fqdn"
  omit_hostname = false
  interval = "10s"
  round_interval = true
  metric_batch_size = 1000
  metric_buffer_limit = 10000
  collection_jitter = "0s"
  flush_interval = "10s"
  flush_jitter = "0s"
  precision = ""
  logfile = ""
  debug = false
  quiet = false

#
# OUTPUTS:
#
[outputs.prometheus_client]
expiration_interval = "60"
listen = ":19100"
metric_version = 2
path = "/metrics"
string_as_label = true


#
# INPUTS:
#
[[inputs.cpu]]
totalcpu = true
[[inputs.prometheus]]
metric_version = 2
url_tag = "ne"
urls = ["http://localhost:19101/metrics"]

```

### System info:

Telegraf 1.16.3 (git: HEAD c7f21728) - works fine
Telegraf 1.17.0 (git: HEAD 3f7a54c9) - Has issues
Red Hat Enterprise Linux Server release 7.9 (Maipo)

### Steps to reproduce:

1. Install telegraf 1.17 and start the service. Observe telegraf log within ~10 seconds the parsing error begins.

### Expected behavior:

Jan 19 13:59:51  systemd[1]: Started The plugin-driven server agent for reporting metrics into InfluxDB.
Jan 19 13:59:51  telegraf[71676]: 2021-01-19T18:59:51Z I! Starting Telegraf 1.16.3
Jan 19 13:59:51  telegraf[71676]: 2021-01-19T18:59:51Z I! Loaded inputs: cpu prometheus
Jan 19 13:59:51  telegraf[71676]: 2021-01-19T18:59:51Z I! Loaded aggregators:
Jan 19 13:59:51  telegraf[71676]: 2021-01-19T18:59:51Z I! Loaded processors:
Jan 19 13:59:51  telegraf[71676]: 2021-01-19T18:59:51Z I! Loaded outputs: prometheus_client
Jan 19 13:59:51  telegraf[71676]: 2021-01-19T18:59:51Z I! Tags enabled: host=testsystem.fqdn ip4_addr=192.168.1.10 job=telegraf
Jan 19 13:59:51  telegraf[71676]: 2021-01-19T18:59:51Z I! [agent] Config: Interval:10s, Quiet:false, Hostname:"testsystem.fqdn", Flush Interval:10s
Jan 19 13:59:51  telegraf[71676]: 2021-01-19T18:59:51Z I! [outputs.prometheus_client] Listening on http://[::]:19100/metrics

### Actual behavior:

Jan 19 13:38:54  telegraf[58933]: 2021-01-19T18:38:54Z I! Starting Telegraf 1.17.0
Jan 19 13:38:54  telegraf[58933]: 2021-01-19T18:38:54Z I! Loaded inputs: cpu prometheus
Jan 19 13:38:54  telegraf[58933]: 2021-01-19T18:38:54Z I! Loaded aggregators:
Jan 19 13:38:54  telegraf[58933]: 2021-01-19T18:38:54Z I! Loaded processors:
Jan 19 13:38:54  telegraf[58933]: 2021-01-19T18:38:54Z I! Loaded outputs: prometheus_client
Jan 19 13:38:54  telegraf[58933]: 2021-01-19T18:38:54Z I! Tags enabled: host=testsystem.fqdn ip4_addr=192.168.1.10 job=telegraf
Jan 19 13:38:54  telegraf[58933]: 2021-01-19T18:38:54Z I! [agent] Config: Interval:10s, Quiet:false, Hostname:"testsystem.fqdn", Flush Interval:10s
Jan 19 13:38:54  telegraf[58933]: 2021-01-19T18:38:54Z I! [outputs.prometheus_client] Listening on http://[::]:19100/metrics
Jan 19 13:39:00  telegraf[58933]: 2021-01-19T18:39:00Z E! [inputs.prometheus] Error in plugin: error reading metrics for http://localhost:19101/metrics: reading text format failed: text format parsing error in line 1: invalid metric name
Jan 19 13:39:10  telegraf[58933]: 2021-01-19T18:39:10Z E! [inputs.prometheus] Error in plugin: error reading metrics for http://localhost:19101/metrics: reading text format failed: text format parsing error in line 1: invalid metric name

The error will repeat every 10 seconds forever.

### Additional info:

This is occuring on multiple systems, all of which are running 1.17. We have not changed the telegraf.conf file since the version upgrade. As you can see the telegraf.conf is managed by puppet and we can ensure it is identical across many systems. If you downgrade telegraf it immediately starts working using the same telegraf.conf. 

As far as I can tell our config file is valid syntax. If i'm mistaken please let me know the issue!

**Comment by debu99** [comment]
had the same issue ....

**Comment by Aladex** [comment]
I think that was fixed in #8366
You need to build a new version from master or wait for the new release

**Comment by sjwang90** [comment]
Should be fixed by #8545
